iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/MoveFeederBay.java

```diff
@@ -87,46 +139,40 @@ public String getName() {
 
     @Override
     public NetworkModificationImpact hasImpactOnNetwork(Network network) {
-        NetworkModificationImpact impact = DEFAULT_IMPACT;
+        impact = DEFAULT_IMPACT;
 
-        // Check connectable and target busbar section exists
+        // Validate connectable and target busOrBusBar section exist
         Connectable<?> connectable = network.getConnectable(connectableId);
-        BusbarSection targetBusbarSection = network.getBusbarSection(targetBusOrBusbarSectionId);
+        BusbarSection targetBusbarSection = network.getBusbarSection(targetBusOrBusBarSectionId);
+
         if (connectable == null || connectable instanceof BusbarSection || targetBusbarSection == null) {
-            impact = NetworkModificationImpact.CANNOT_BE_APPLIED;
-            return impact;
+            return NetworkModificationImpact.CANNOT_BE_APPLIED;
```

Comment:
You have to keep the affectation to `impact` (the attribute from the parent class).